src/modules/jobs/job.service.ts

```diff
@@ -211,4 +212,39 @@ export class JobsService {
       throw new InternalServerErrorException();
     }
   }
+
+  async updateJob(payload: UpdateJobDto, user: UserDto): Promise<void> {
+    try {
+      const { id, skills, ...jobPayload } = payload;
+      const job = await this.findOne({ id });
+      if (!(job && job.owner.id === user.id)) {
```

Comment:
I see this condition 2d or 3d times. and I don't understand it, if you have a relationship connection how can you have this error? why do you check for specific this error